### PR TITLE
Enable nullability in ListViewGroupCollection and ListViewGroupItemCollection

### PR DESCRIPTION
**Pull request by gpetrou**
## Proposed changes

- Enable nullability in ListViewGroupCollection and ListViewGroupItemCollection.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/6909)

### CODE REVIEW

File: src/System.Windows.Forms/src/System/Windows/Forms/ListViewGroupItemCollection.cs [hunk 1]
```diff
@@ -20,22 +18,22 @@ public ListViewGroupItemCollection(ListViewGroup group)
...
+        private ArrayList Items => _items ??= new ArrayList();
 
         public bool OwnerIsVirtualListView => _group.ListView is not null && _group.ListView.VirtualMode;
 
         public bool OwnerIsDesignMode => _group.ListView?.Site is not null && _group.ListView.Site.DesignMode;
 
         public ListViewItem this[int index]
         {
-            get => (ListViewItem)Items[index];
+            get => (ListViewItem)Items[index]!;
```
Comment (RussKie) [inline src/System.Windows.Forms/src/System/Windows/Forms/ListViewGroupItemCollection.cs:29]: 💭  A possible solution to `!` operator is to convert `ArrayList` to `List<ListViewItem>`. 

The only issue with this is the following method:
```diff
- public void CopyTo(Array dest, int index) => Items.CopyTo(dest, index);
+ public void CopyTo(Array dest, int index) => Array.Copy(Items.ToArray(), dest, index);
```
...which makes an extra allocation... 🤔 

@stephentoub, I'm curious - are there any ways to convert a `List<T>` to an array without extra allocations or resorting to a reflection? Thanks

Comment (stephentoub) [reply src/System.Windows.Forms/src/System/Windows/Forms/ListViewGroupItemCollection.cs:29]: > are there any ways to convert a List<T> to an array without extra allocations or resorting to a reflection?

To an array? No.  But you can access `List<T>`'s backing array as a span via https://docs.microsoft.com/en-us/dotnet/api/system.runtime.interopservices.collectionsmarshal.asspan?view=net-6.0#system-runtime-interopservices-collectionsmarshal-asspan-1(system-collections-generic-list((-0))).

Comment (stephentoub) [reply src/System.Windows.Forms/src/System/Windows/Forms/ListViewGroupItemCollection.cs:29]: > The only issue with this is the following method

`List<T>` implements the non-generic `ICollection`, so such a method would presumably just be:
```C#
public void CopyTo(Array dest, int index) => ((ICollection)Items).CopyTo(dest, index);
```

**Review by RussKie — APPROVED** [review]

File: src/System.Windows.Forms/src/System/Windows/Forms/ListViewGroupItemCollection.cs (hunk 1 above)
Comment (RussKie) [reply src/System.Windows.Forms/src/System/Windows/Forms/ListViewGroupItemCollection.cs]: Thank you!

**Pull request merged**